oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoder.java

```diff
@@ -280,9 +300,16 @@ public JwkSetUriJwtDecoderBuilder restOperations(RestOperations restOperations)
 			}
 		}
 
+		JWKSource<SecurityContext> jwkSource(ResourceRetriever resourceRetriever) {
+			return Optional.ofNullable(this.cache)
```

Comment:
As awesome as `Optional` is, it's too slow to be used in framework code. Would you please use a plain if statement instead?